tests/end-to-end/run.sh

```diff
@@ -0,0 +1,262 @@
+#!/bin/bash
+
+#
+# Copyright (c) 2020, NVIDIA CORPORATION.  All rights reserved.
+#
+# NVIDIA CORPORATION and its licensors retain all intellectual property
+# and proprietary rights in and to this software, related documentation
+# and any modifications thereto.  Any use, reproduction, disclosure or
+# distribution of this software and related documentation without an express
+# license agreement from NVIDIA CORPORATION is strictly prohibited.
+#
+
+
+set -e
+env | grep "PATH"
+
+# End-to-end testing script.
+
+verify_diff () {
+    if [ "$3" == "h5" ]
+    then
+        h5diff $1 $2
+    else
+	diff $1 $2
+    fi
+
+    if [ $? != 0 ]
+    then
+        echo "Files $1 and $2 do not match."
+        echo "Abort test"
+        exit
+    fi	
+
+}
+
+echo ""
+echo "Step 0: Initialize environment..."
+echo ""
+test_dir=$(readlink -f $(dirname "$0"))
+test_root_dir=$(readlink -f "$test_dir/..")
+root_dir=$(readlink -f "$test_root_dir/..")
+data_dir="$test_root_dir/data/end-to-end"
+ref_dir="$test_root_dir/reference/end-to-end"
+out_dir="$test_root_dir/result/end-to-end"
+expected_results_dir="$test_root_dir/expected_results/end-to-end"
+
+saved_model_dir="$test_root_dir/data/end-to-end/pretrained_models"
+config_dir="$root_dir/configs"
+
+# Switch to root directory before running script.
+cd $test_root_dir
+
+if [ -d "$out_dir" ]; then
+    rm -rf $out_dir
+fi
+mkdir -p $out_dir
+
+
+echo ""
```

Comment:
let's break each of these steps into separate test.sh scripts with descriptive names. so the individual scripts would look like this - 

```
preamble (a shared setup function to set basic directories)
single test, e.g. Split the given regions of the genome into train, val, and holdout/test intervals...
verify diff and output success or failure
```

Reply:
done.